File: java_pro/src/main/java/jv/com/tencent/cain/design_pattern/simple_factory/AddOperation.java
package jv.com.tencent.cain.design_pattern.simple_factory;

/**
 * @author cainjiang
 * @date 2019-02-21
 * 加法操作类
 */
public class AddOperation extends Operation {

    @Override
    public double getResult() {
        return numA + numB;
    }
}

File: java_pro/src/main/kotlin/kt/com/tencent/cain/desigin_pattern/simple_factoty/AddOperation.kt
package kt.com.tencent.cain.desigin_pattern.simple_factoty

/**
 * @author cainjiang
 * @date 2019-02-22
 * 加法操作类
 */
class AddOperation : Operation() {
    override fun getResult(): Double {
        return numA + numB
    }
}